OpenEdXMobile/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -135,9 +137,12 @@ public static void setAuthorText(@NonNull TextView textView, @NonNull IAuthorDat
 
     private static void applyColorAndStyle(@NonNull Context context,
                                            @NonNull SpannableString span,
+                                           boolean isAuthorAnonymous,
                                            @ColorRes int color, int style) {
-        span.setSpan(new ForegroundColorSpan(context.getResources().getColor(color)), 0,
-                span.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        if (!isAuthorAnonymous) {
+            span.setSpan(new ForegroundColorSpan(context.getResources().getColor(color)), 0,
+                    span.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        }
         span.setSpan(new StyleSpan(style), 0, span.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
     }
```

Comment:
I would actually get rid of this method now. It's only called in two places, and since in one of them the author is never anonymous, the actual utility is only being provided to one place, thus removing it's justification as a utility method. Plus, with the addition of the `isAuthorAnonymous` parameter, it's no longer a generic method, thus decreasing it's usefulness sharply.